### PR TITLE
Standard

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/codeforscience/sciencefair/issues/140

@blahah I decided to break up this into many smaller commits so it's easier to follow along. Instead of having one mega diff. Also easier to discuss separate commits.

Obviously open to discussion on matters that you don't agree with. We can keep this PR open and add some tweaks on top, then squash.

Cheers

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,5 +40,10 @@
     "rimraf": "^2.5.2",
     "standard": "^10.0.2",
     "temporary": "^0.0.8"
+  },
+  "standard": {
+    "ignore": [
+      "app/client/lib"
```
Comment (ralphtheninja) [inline package.json]: Ignoring this for now, but something we might want to address as well? I'm unsure how the lens code is built/managed etc.

File: app/client/keybindings.js [hunk 2]
```diff
@@ -2,7 +2,6 @@ const Mousetrap = require('mousetrap')
 const {BrowserWindow} = require('electron').remote
 
 module.exports = (state, bus) => {
-
```
Comment (ralphtheninja) [inline app/client/keybindings.js]: This commits are changes made by `standard --fix` with no manual changes.

File: app/client/models/collection.js [hunk 3]
```diff
@@ -51,14 +55,9 @@ module.exports = (state, bus) => {
...
       activescan = null
-      if (err) {
-        if (err.message == 'premature close') {
-          // activescan was destroyed
-
-        } else {
-          throw err
-        }
-      } else {
+      if (err && !_activeScanDestroyedError(err)) {
```
Comment (ralphtheninja) [inline app/client/models/collection.js]: I thought this was a bit cleaner and more readable.

File: app/client/models/error.js [hunk 4]
```diff
@@ -1 +1 @@
-module.exports = (state, bus) => bus.on('error', err => { throw error })
```
Comment (ralphtheninja) [inline app/client/models/error.js]: `error` not defined

File: app/client/models/results.js [hunk 5]
```diff
@@ -22,7 +22,7 @@ module.exports = (state, bus) => {
   }
 
   const checkfiles = p => p.filesPresent(
-    (err, progress, updated) => { if (updated) render() }
+    (err, progress, updated) => { if (!err && updated) render() }
```
Comment (ralphtheninja) [inline app/client/models/results.js]: Not really handling the error, but standard complains if `err` is not referenced.

File: app/run.js [hunk 6]
```diff
@@ -12,64 +12,64 @@ const shouldQuit = app.makeSingleInstance((cliargs, cwd) => {
 })
 
 if (shouldQuit) {
-  return app.quit()
-}
+  app.quit()
```
Comment (ralphtheninja) [inline app/run.js]: I know, this is not ideal :) I also like to bail out early if possible. Also helps with readability with less scope. We could add a `// eslint-disable-line` here as well and call it a day.

File: app/client/views/overlay_confirm.js [hunk 7]
```diff
@@ -62,14 +62,14 @@ module.exports = (msg, cb) => {
...
-    cb(true)
+    cb(true) // eslint-disable-line
   }
 
   const nobtn = html`<div class="${style.button}">NO</div>`
   nobtn.onclick = e => {
     e.preventDefault()
     destroy()
-    cb(false)
+    cb(false) // eslint-disable-line
```
Comment (ralphtheninja) [inline app/client/views/overlay_confirm.js]: Can these callbacks be of the form `(err, result)` instead?

File: app/client/models/error.js (hunk 4 above)
Comment (blahah) [inline app/client/models/error.js]: o_O

...yup, linting ftw

**Comment by ralphtheninja** [comment]
@blahah Any changes you think I should make? Guess you can squash on github when merging the PR (haven't tried it myself).

File: app/client/models/error.js (hunk 4 above)
Comment (ralphtheninja) [inline app/client/models/error.js]: Finding stuff like this with a linter is really nice indeed. Also very nice to get rid of unused stuff.

**Comment by ralphtheninja** [comment]
beep boop?

**Comment by blahah** [comment]
@ralphtheninja sorry, i had some life stuff that got very in the way. I'm happy to merge this :)

**Pull request merged**